server/src/main/java/org/opensearch/cluster/service/applicationtemplates/SystemTemplatesService.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service.applicationtemplates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.ClusterChangedEvent;
+import org.opensearch.cluster.ClusterStateListener;
+import org.opensearch.cluster.LocalNodeClusterManagerListener;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Service class to orchestrate execution around available templates' management.
+ */
+public class SystemTemplatesService implements LocalNodeClusterManagerListener {
+
+    public static final int APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD_DEFAULT_COUNT = 50;
+
+    public static final Setting<Integer> SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD = Setting.intSetting(
+        "cluster.application_templates.load",
+        APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD_DEFAULT_COUNT,
+        0,
+        1000,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final List<SystemTemplatesPlugin> systemTemplatesPluginList;
+
+    private final Settings settings;
+
+    private final AtomicBoolean loaded = new AtomicBoolean(false);
+
+    private volatile int templatesToLoad = APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD_DEFAULT_COUNT;
+
+    private static final Logger logger = LogManager.getLogger(SystemTemplatesService.class);
+
+    public SystemTemplatesService(List<SystemTemplatesPlugin> systemTemplatesPluginList,
+                                  ClusterSettings clusterSettings,
+                                  Settings settings) {
+        this.systemTemplatesPluginList = systemTemplatesPluginList;
+        clusterSettings.addSettingsUpdateConsumer(SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD, this::setTemplatesToLoad);
+        this.settings = settings;
+    }
+
+    @Override
+    public void onClusterManager() {
+        refreshTemplates();
+    }
+
+    @Override
+    public void offClusterManager() {
+        // do nothing
+    }
+
+    private void refreshTemplates() {
+        if (loaded.compareAndSet(false, true)) {
+            if (SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD.get(settings) > 0) {
+                int countOfTemplatesToLoad = templatesToLoad;
+                int templatesLoaded = 0;
+                int failedLoadingTemplates = 0;
+                int failedLoadingRepositories = 0;
+
+                for (SystemTemplatesPlugin plugin : systemTemplatesPluginList) {
+                    try (TemplateRepository repository = plugin.loadRepository()) {
```

Comment:
SystemTemplatesRepository is not mandatory? as we swallowing the failures

Reply:
Since there can be multiple repositories and multiple templates, I was passing forward by ignoring failures due to loading of 1 repo and move on to the next.